lib/udap_security_test_kit/client_suite/token_request_verification.rb

```diff
@@ -174,5 +188,36 @@ def extract_token_from_response(request)
     rescue StandardError
       nil
     end
+
+    def check_refresh_request_params(oauth_flow, params, request_num)
+      if oauth_flow == CLIENT_CREDENTIALS_TAG
+        add_message('error',
+                    "Invalid refresh request #{request_num} found during client_credentials flow.")
+        return
+      end
+
+      if params['grant_type'] != 'refresh_token'
+        add_message('error',
+                    "Refresh request #{request_num} had an incorrect `grant_type`: expected 'refresh_token', " \
+                    "but got '#{params['grant_type']}'")
+      end
+      if params['client_assertion_type'] != 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer'
+        add_message('error',
+                    "Confidential asymmetric refresh request #{request_num} had an incorrect " \
+                    "`client_assertion_type`: expected 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer', " \
+                    "but got '#{params['client_assertion_type']}'")
+      end
+
+      authorization_code = MockUDAPServer.refresh_token_to_authorization_code(params['refresh_token'])
+      authorization_request = MockUDAPServer.authorization_request_for_code(authorization_code, test_session_id)
+      if authorization_request.present?
+        # TODO: - check that the scope is a subset of the original authorization code request
```

Comment:
Is this supposed to be done as part of this PR?

Reply:
This seems potentially quite tricky to do, so i was not planning on doing it in this PR. I know there are different schools of thought about where this kind of todo should go. I created [FI-4045](https://oncprojectracking.healthit.gov/support/browse/FI-4045) to track this if you think it should live only in jira.